### ISSUE TITLE
Non-trainable layers are updated during fine-tuning

### ISSUE THREAD

**Issue by dubchek** [tensorflow/models#1452]
### System information
- **What is the top-level directory of the model you are using**: models-master\slim
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: I haven't written my code. I've only applied  https://github.com/tensorflow/models/pull/1422 and https://github.com/tensorflow/models/pull/1346 fixes which are related to reading of the datasets
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows Server 2016 (Google Cloud Platform)
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.1.0
- **Bazel version (if compiling from source)**: I didn't use bazel
- **CUDA/cuDNN version**: 8.0.61/5.1
- **GPU model and memory**: 1 x NVIDIA Tesla K80, 15GB RAM
- **Exact command to reproduce**: 

Checkpoint is downloaded and unzipped manually.

**set work_dir=C:\Dev\work_dir
set train_dir=%work_dir%\train_dir
set datasets=%work_dir%\datasets
set checkoints=%work_dir%\checkpoints
set curr_model=inception_resnet_v2
set curr_ckpt_date=2016_08_30
python train_image_classifier.py ^
--train_dir=%train_dir%\%curr_model% ^
--dataset_dir=%datasets%\%curr_dataset% ^
--dataset_name=%curr_dataset% ^
--dataset_split_name=train ^
--model_name=%curr_model% ^
--checkpoint_path=%checkpoints%\%curr_model%\%curr_model%_%curr_ckpt_date%.ckpt ^
--checkpoint_exclude_scopes=InceptionResnetV2/Logits,InceptionResnetV2/AuxLogits ^
--trainable_scopes=InceptionResnetV2/Logits,InceptionResnetV2/AuxLogits**

So the resolved command looks like:

**python train_image_classifier.py ^
--train_dir=C:\Dev\work_dir\train_dir\inception_resnet_v2 ^
--dataset_dir=C:\Dev\work_dir\datasets\flowers ^
--dataset_name=flowers ^
--dataset_split_name=train ^
--model_name=inception_resnet_v2 ^
--checkpoint_path=C:\Dev\work_dir\checkpoints\inception_resnet_v2\inception_resnet_v2_2016_08_30.ckpt ^
--checkpoint_exclude_scopes=InceptionResnetV2/Logits,InceptionResnetV2/AuxLogits ^
--trainable_scopes=InceptionResnetV2/Logits,InceptionResnetV2/AuxLogits**

### Describe the problem
I fine-tuned inception-resent-v2 model on flowers dataset. I applied appropriate flags:
--checkpoint_exclude_scopes=InceptionResnetV2/Logits,InceptionResnetV2/AuxLogits ^
--trainable_scopes=InceptionResnetV2/Logits,InceptionResnetV2/AuxLogits
I expected that only Logits and AuxLogits blocks will be trained and all other layers will not be changed during the fine-tuning. However after training I have found that Tensorboard displays changes for BatchNormalization and Convolutional weights (histograms ->InceptionResnetV2 section) for such blocks as Repeat, Repeat2, Block8 etc. From my point of view it is a bug because those layers should be untouched during the fine-tuning but they are.
Could you please review this issue?

### Source code / logs
Sources:
https://drive.google.com/open?id=0B71CmqrSAIWWNGhVUG43Ulc5UFk

Original checkpoint:
https://drive.google.com/open?id=0B71CmqrSAIWWNGhVUG43Ulc5UFk

Training directory:
https://drive.google.com/open?id=0B71CmqrSAIWWSndmSnVzaUptcGM

Screenshot of changed non-trainable layers weights:
![screen shot 2017-05-07 at 1 56 39 pm](https://cloud.githubusercontent.com/assets/17811772/25780326/a7e5503c-332e-11e7-8681-684fb310a3ae.jpg)

**Comment by skye** [comment]
@shlens could you take a look or redirect to someone who can? Thanks!

**Comment by shlens** [comment]
@sguada , would you be able to take a look?

**Comment by pronot** [comment]
I observed the same situation. Any progress?

**Comment by neouyghur** [comment]
I also observed the same situation. Did someone find the problem?

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
